src/util/http/websocket/MessageSender.cpp

```diff
@@ -0,0 +1,46 @@
+//   Copyright 2023, University of Freiburg,
+//   Chair of Algorithms and Data Structures.
+//   Author: Robin Textor-Falconi <textorr@informatik.uni-freiburg.de>
+
+#include "util/http/websocket/MessageSender.h"
+
+namespace ad_utility::websocket {
+
+MessageSender::MessageSender(DistributorWithFixedLifetime distributor,
+                             net::any_io_executor executor)
+    : distributor_{std::move(distributor),
+                   [executor](auto&& distributor) {
+                     auto coroutine =
+                         [](auto distributor) -> net::awaitable<void> {
+                       co_await distributor.distributor_->signalEnd();
+                     };
+                     net::co_spawn(executor, coroutine(AD_FWD(distributor)),
+                                   net::detached);
+                   }},
+      executor_{std::move(executor)} {}
```

Comment:
I like this solution now, but it has to be explained here:
The distributor is moved into the awaitable, so the `OwningQueryId` which is part of it is only destroyed after the call to `signalEnd` is actually completed and the coroutine is destroyed. That is so implicit that it needs a comment, but again, I like the solution.

Reply:
And also add here that `signalEnd` erases from the queryHub and THEN the owning query Id is unregistered.